subprojects/testing-base/src/test/groovy/org/gradle/api/internal/tasks/testing/processors/JULRedirectorTest.groovy

```diff
@@ -0,0 +1,144 @@
+    private final Logger logger1 = Logger.getLogger("")
+    private final Logger logger2 = Logger.getLogger("hello")
+    private final Logger logger3 = Logger.getLogger("hello.world")
+    private static String savedJULConfigClassProperty = null;
+
+    public static class DummyFormatter extends Formatter {
+        @Override
+        String format(LogRecord record) {
+            return "[" + record.level + "] " + record.message + EOL
+        }
+    }
+
+    public static class JULCustomInit {
+        JULCustomInit() {
+            LogManager.getLogManager().readConfiguration(org.apache.commons.io.IOUtils.toInputStream(
+                'handlers = java.util.logging.ConsoleHandler' + EOL +
+                    '.level = FINEST' + EOL +
+                    'java.util.logging.SimpleFormatter.format = test' + EOL +
+                    'java.util.logging.ConsoleHandler.formatter = ' + DummyFormatter.class.name + EOL +
+                    'java.util.logging.ConsoleHandler.level = FINEST' + EOL +
+                    'hello.level = INFO' + EOL +
+                    'hello.world.level = FINER' + EOL))
+        }
+    }
+
+    def setup() {
+        savedJULConfigClassProperty = System.getProperty("java.util.logging.config.class")
+    }
+
+    def cleanup() {
+        if (savedJULConfigClassProperty == null) {
+            System.clearProperty("java.util.logging.config.class")
+        } else {
+            System.setProperty("java.util.logging.config.class", savedJULConfigClassProperty)
+        }
+    }
+
+    def "start and stop do nothing when nothing is redirected"() {
+        when:
+        redirector.start()
+        [Level.SEVERE, Level.WARNING, Level.INFO, Level.CONFIG, Level.FINE, Level.FINER, Level.FINEST].each {
+            logger1.log(it, "Test");
+            logger2.log(it, "Test");
+            logger3.log(it, "Test");
+        }
+        redirector.stop()
+
+        then:
+        System.out == outputs.stdOutPrintStream
+        System.err == outputs.stdErrPrintStream
+    }
+
+    def "start and stop output with redirection and default logging"() {
+        when:
+        redirector.redirectStandardOutputTo(stdOutListener)
+        redirector.redirectStandardErrorTo(stdErrListener)
+
+        redirector.start()
+        [Level.SEVERE, Level.WARNING, Level.INFO, Level.CONFIG, Level.FINE, Level.FINER, Level.FINEST].each {
+            logger1.log(it, "Test");
+            logger2.log(it, "Test");
+            logger3.log(it, "Test");
+        }
+        redirector.stop()
+
+        then:
+        3 * stdErrListener.onOutput('SEVERE: Test' + EOL)
+        3 * stdErrListener.onOutput('WARNING: Test' + EOL)
+        3 * stdErrListener.onOutput('INFO: Test' + EOL)
+        0 * stdErrListener.onOutput('CONFIG: Test' + EOL)
+        0 * stdErrListener.onOutput('FINE: Test' + EOL)
+        0 * stdErrListener.onOutput('FINER: Test' + EOL)
+        0 * stdErrListener.onOutput('FINEST: Test' + EOL)
```

Comment:
I will add strict mocking in the "start and stop output with redirection and custom logging" test, but not here.  Here the output is defined by the default SimpleFormat settings, which include non-deterministic things (e.g., time the test is being executed).

In the other test, the logging output is fully controlled, so I have no objection make checking more strict.